pkg/util/oidcbuilder/jwks.go

```diff
@@ -0,0 +1,136 @@
+package oidcbuilder
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"gopkg.in/square/go-jose.v2"
+)
+
+func CreateKeyPair() (encPrivateKey []byte, encPublicKey []byte, err error) {
+	bitSize := 4096
```

Comment:
is this the bit size & recommendation of OpenShift for SA private keys?

Reply:
Creation of jwks is mostly copied from [Cloud-credential-operator](https://github.com/openshift/cloud-credential-operator/blob/master/pkg/cmd/provisioning/create_key_pair.go#L35) same as [ROSA STS](https://gitlab.cee.redhat.com/service/uhc-clusters-service/-/tree/master/pkg/aws/cloudcredentialbuilder)

Reply:
4096 should be ok for FIPS compliance as well: https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-56Br2.pdf

Reply:
Also - would it be possible to have the cloud credential operator folks publish functions for generating key pairs, so that way ROSA and ARO can consume those directly instead of repeating the same code?

Reply:
@niontive they do actually: https://github.com/openshift/cloud-credential-operator/blob/master/docs/ccoctl.md#creating-rsa-keys-1 It's a CLI tool though, it's a bit different for us since we own the OIDC stuff.